File: TypeScript/Code/Module/I18N/I18NConfigCategory.ts
import { Log } from "../../../Mono/Module/Log/Log";


export class I18NConfig {
    /** Id */
    public id: number
    /** 内容  */
    public value: string
}

export class I18NConfigCategory{

    private dict = new Map<number, I18NConfig>();

    private list:I18NConfig[] = [];

    public endInit()
    {
        for(let i =0 ;i<this.list.length;i++)
        {
            const config:I18NConfig = this.list[i];

            this.dict.set(config.id, config);
        }            
    }
    
    public get(id: number): I18NConfig
    {
        let item: I18NConfig = this.dict.get(id);
        
        if (item == null)
        {
            Log.error("配置找不到，配置表名: I18NConfig，配置id: "+id);
        }

        return item;
    }

    public contain(id: number): boolean
    {
        return this.dict.has(id);
    }

    public getAll(): Map<number, I18NConfig>
    {
        return this.dict;
    }

    public getAllList(): I18NConfig[]
    {
        return this.list;
    }
}

File: Tools/ExcelExport/Helper/I18NConfigCategory.cs
using System;
using System.Collections.Generic;


public class I18NConfigCategory
{
    public List<I18NConfig> list = new List<I18NConfig>();

    public List<I18NConfig> GetAllList()
    {
        return this.list;
    }
}

public partial class I18NConfig
{
    /// <summary>Id</summary>
    public int id { get; set; }

    /// <summary>索引标识</summary>
	public string key { get; set; }

    /// <summary>内容</summary>
    public string value { get; set; }
}

